### ISSUE TITLE
CHALLENGE: JavaScript functionality

### ISSUE THREAD

**Issue by JohnHasthorpe** [tedrick/Survey123-Testing#144]
We have added support for extending the functionality of your surveys using JavaScript.
Here is a workflow - test it out and then have a go with your own surveys and JS.

**Testing the survey**
1) In Connect publish the attached survey. 
2) Take a look at the following feature [Service](https://services1.arcgis.com/e7dVfn25KpfE6dDd/arcgis/rest/services/Extent/FeatureServer/0)
This is a feature service containing a polygon extent (the survey limit)
3) In the survey directory, create a directory called 'extensions'
4) Copy the attached .js file into the extensions directory
5) Close and re-open connect
6) In Connect test the survey by trying to submit a survey out side of the extent (note that you wont see the polygon in the map) - the survey constrant should prevent you from submitting.

**Understanding the survey**
The constraint is based on the value returned by the JS function. If the function returns the value "No Features Found" (i.e the collected point does not intersect the extent polygon) then the survey can not be submitted.

Take a look at the data1 question and the calculation it contains:
pulldata("@javascript","myFunctions.js","queryFeatureLayerByLocationScalar",${myURL},string(${location}),"*",pulldata("@property","token"),true)

We are using the puldata @javascript function, with a number paramtes:
- _myFunctions.js_  -  The name of the JS file
- _queryFeatureLayerByLocationScalar_ - The name of the function
- _${myURL},string(${location}),"*",pulldata("@property","token"),true_ - The parameters that need to be passed into the function: service url, location string, returned fields, token, debug mode (boolean).

[Redlands Zoning.xlsx](https://github.com/tedrick/Survey123-Testing/files/2331824/Redlands.Zoning.xlsx)
[myFunctions.zip](https://github.com/tedrick/Survey123-Testing/files/2331828/myFunctions.zip)

**Comment by WaltersGareth** [comment]
The Connect application doesn't refresh the code if you edit the function while Connect is open.

I wrote:
`function helloWorld(){
	return "JavaScript functions rock!";
}`

and the app shows the correct text. Then while the app is open and I changed the return value, pressed the refresh button but the form maintained the original return result.

It only refreshed to the new value if I restart Survey123 Connect.
